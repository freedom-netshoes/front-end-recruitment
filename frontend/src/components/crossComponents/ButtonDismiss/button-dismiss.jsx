import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import CSSModules from 'react-css-modules'

import styles from './button-dismiss.less'

@CSSModules(styles)
export default class ButtonDismiss extends Component {
  static defaultProps = {
    clickButton: null,
    mouseEnter: null,
    mouseLeave: null,
    extraCssClass: ''
  };

  static propTypes = {
    clickButton: PropTypes.func,
    mouseEnter: PropTypes.func,
    mouseLeave: PropTypes.func,
    extraCssClass: PropTypes.string
  };

  render () {
    return (
      <span {...{
        onClick: this.props.clickButton,
        onMouseEnter: this.props.mouseEnter,
        onMouseLeave: this.props.mouseLeave,
        className: `button ${this.props.extraCssClass}`
      }} />
    )
  }
}
