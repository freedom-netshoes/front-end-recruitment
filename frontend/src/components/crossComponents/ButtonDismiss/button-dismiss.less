@import url('../../../styles/_less-variables.less');
@import url('../../../styles/_less-mixins.less');
:global {
    .button {
        cursor: pointer;
        display: block;
        position: relative;
        top: -17px;
        width: 17px;
        &:before {
            color: @black;
            content: "\2716";
            font-weight: bolder;
        }
    }
    .button:hover {
        &:before {
            color: @white;
        }
    }
}