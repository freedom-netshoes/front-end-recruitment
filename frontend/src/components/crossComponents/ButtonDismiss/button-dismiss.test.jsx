/* global describe, it, expect */
import React from 'react'
import ButtonDismiss from './button-dismiss'
import { shallow } from 'enzyme'

describe('Testing <ButtonDismiss /> component', () => {
  it('renders the ButtonDismiss component', () => {
    const component = shallow(<ButtonDismiss />)
    expect(component).toHaveLength(1)
  })

  it('Testing <ButtonDismiss /> snapshop', () => {
    const component = shallow(<ButtonDismiss />)
    expect(component).toMatchSnapshot()
  })
})
