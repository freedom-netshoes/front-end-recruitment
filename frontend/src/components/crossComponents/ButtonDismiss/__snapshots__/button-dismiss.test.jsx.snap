// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ButtonDismiss /> component Testing <ButtonDismiss /> snapshop 1`] = `
<span
  className="button "
  onClick={null}
  onMouseEnter={null}
  onMouseLeave={null}
/>
`;
